### PR TITLE
Fix RecSys AddressSanitizer https://github.com/pytorch/glow/issues/4792

### LINKED ISSUE #4792: RecommendationSystemTest hits AddressSanitizer issue with static placeholders enabled.
Commit https://github.com/pytorch/glow/commit/1a9acc292e8ea88627eeeef3572510dd9c7daa0b updated the RecommendationSystemTest to support using static placeholders.

There are two cases (see below) where the deferred weight loader is using references to tensors that have been allocated on the stack after the tensors have gone out of scope resulting in an asan failure.

https://github.com/pytorch/glow/blob/79ed8736f9ef3ae10dabd1695cfa4e8665eb08f6/tests/unittests/RecommendationSystemTest.cpp#L613
https://github.com/pytorch/glow/blob/79ed8736f9ef3ae10dabd1695cfa4e8665eb08f6/tests/unittests/RecommendationSystemTest.cpp#L704

### PR DESCRIPTION
**Pull request by omromano**
Summary:
Changed the implementation to use a tensor that is held by the TestDeferredWeightLoader instead of using a reference to a local var.
 
Documentation:

Fixes #4792

### CODE REVIEW

**Comment by omromano** [comment]
Hi @jfix71, can you please take a look.

**Comment by jfix71** [comment]
Really sorry for the major delays here, I've been super swamped. I will definitely get to reviewing next week.

File: tests/unittests/RecommendationSystemTest.cpp [hunk 1]
```diff
@@ -604,17 +610,15 @@ class RecommendationSystemTest : public BackendTest {
...
               mod, {embSizes[i], embDim}, "data" + std::to_string(i),
               useFP16SLWS);
-          auto tensor = Tensor(ph->getType());
 
           ph->setStatic(true);
-          tensor.getHandle<uint8_t>().randomize(UINT8_MIN, UINT8_MAX,
-                                                mod.getPRNG());
-
-          loader.addWeight(&tensor);
+          auto tensor = loader.addWeight(ph->getType());
```
Comment (jfix71) [inline tests/unittests/RecommendationSystemTest.cpp:615]: nit: `auto *tensor`

File: tests/unittests/RecommendationSystemTest.cpp [hunk 2]
```diff
@@ -695,17 +699,15 @@ class RecommendationSystemTest : public BackendTest {
           Placeholder *ph = createFusedRowwiseQuantizedPlaceholder(
               mod, {tableSizes[i], embeddingDim}, "data" + std::to_string(i),
               useFP16SLWS);
-          auto tensor = Tensor(ph->getType());
-
           ph->setStatic(true);
-          tensor.getHandle<uint8_t>().randomize(UINT8_MIN, UINT8_MAX,
-                                                mod.getPRNG());
+          auto tensor = loader.addWeight(ph->getType());
```
Comment (jfix71) [inline tests/unittests/RecommendationSystemTest.cpp:703]: nit: `auto *tensor`

**Review by jfix71 — APPROVED** [review]
👍

**Pull request closed without merging**